packages/next/server/next-server.ts

```diff
@@ -811,7 +824,13 @@ export default class Server {
 
               parsedDestination.search = stringifyQuery(req, query)
 
-              const updatedDestination = formatUrl(parsedDestination)
+              let updatedDestination = formatUrl(parsedDestination)
+
+              if (updatedDestination.startsWith('/')) {
```

Comment:
Should this also check for backslashes?

Reply:
The destination can't start with a backslash here as it would be formatted away and we only want to apply normalizing for non-external redirects here.